hi-ml/src/health_ml/runner.py

```diff
@@ -0,0 +1,361 @@
+    :param model_deployment_hook: an optional function for deploying a model in an application-specific way.
+    If present, it should take a model config, and an AzureML Model as arguments, and return an optional
+    Path and a further object of any type.
+    :param command_line_args: command-line arguments to use; if None, use sys.argv.
+    """
+
+    def __init__(self,
+                 project_root: Path,
+                 yaml_config_file: Path):
+        self.project_root = project_root
+        self.yaml_config_file = yaml_config_file
+        # self.post_cross_validation_hook = post_cross_validation_hook
+        # self.model_deployment_hook = model_deployment_hook
+        # model_config and experiment_config are placeholders for now, and are set properly when command line args are
+        # parsed.
+        self.model_config: Optional[GenericConfig] = None
+        self.experiment_config: ExperimentConfig = ExperimentConfig()
+        self.lightning_container: LightningContainer = None  # type: ignore
+        # This field stores the MLRunner object that has been created in the most recent call to the run() method.
+        self.ml_runner: Optional[MLRunner] = None
+
+    def parse_and_load_model(self) -> ParserResult:
+        """
+        Parses the command line arguments, and creates configuration objects for the model itself, and for the
+        Azure-related parameters. Sets self.experiment_config and self.model_config to their proper values. Returns the
+        parser output from parsing the model commandline arguments.
+        If no "model" argument is provided on the commandline, self.model_config will be set to None, and the return
+        value is None.
+
+        :return: ParserResult object containing args, overrides and settings
+        """
+        # Create a parser that will understand only the args we need for an AzureConfig
+        parser1 = create_runner_parser()
+        parser_result = parse_args_and_add_yaml_variables(parser1,
+
+                                                          fail_on_unknown_args=False)
+        experiment_config = ExperimentConfig(**parser_result.args)
+        self.experiment_config = experiment_config
+        # self.azure_config = azure_config
+        self.model_config = None
+        if not experiment_config.model:
+            raise ValueError("Parameter 'model' needs to be set to specify which model to run.")
+        model_config_loader: ModelConfigLoader = ModelConfigLoader(**parser_result.args)
+        # Create the model as per the "model" commandline option. This can return either a built-in config
+        # of type DeepLearningConfig, or a LightningContainer.
+        config_or_container = model_config_loader.create_model_config_from_name(model_name=experiment_config.model)
+
+        def parse_overrides_and_apply(c: object, previous_parser_result: ParserResult) -> ParserResult:
+            assert isinstance(c, GenericConfig)
```

Comment:
as written above, can we retire GenericConfig, and just check if something derives from params.Parameterized?